Add label: ClearTouchCollage

duderin0 opened the pull request:
Result:
admin in ~/Documents/GitHub/Installomator/build on cleartouch-label λ sudo ./Installomator.sh cleartouchcollage DEBUG=0
2023-05-01 15:41:10 : INFO  : cleartouchcollage : setting variable from argument DEBUG=0
2023-05-01 15:41:10 : REQ   : cleartouchcollage : ################## Start Installomator v. 10.4beta, date 2023-05-01
2023-05-01 15:41:10 : INFO  : cleartouchcollage : ################## Version: 10.4beta
2023-05-01 15:41:10 : INFO  : cleartouchcollage : ################## Date: 2023-05-01
2023-05-01 15:41:10 : INFO  : cleartouchcollage : ################## cleartouchcollage
2023-05-01 15:41:16 : INFO  : cleartouchcollage : BLOCKING_PROCESS_ACTION=tell_user
2023-05-01 15:41:16 : INFO  : cleartouchcollage : NOTIFY=success
2023-05-01 15:41:16 : INFO  : cleartouchcollage : LOGGING=INFO
2023-05-01 15:41:16 : INFO  : cleartouchcollage : LOGO=/System/Applications/App Store.app/Contents/Resources/AppIcon.icns
2023-05-01 15:41:16 : INFO  : cleartouchcollage : Label type: pkgInZip
2023-05-01 15:41:16 : INFO  : cleartouchcollage : archiveName: Collage.zip
2023-05-01 15:41:16 : INFO  : cleartouchcollage : no blocking processes defined, using Collage as default
2023-05-01 15:41:16 : INFO  : cleartouchcollage : No version found using packageID com.cvte.cleartouch.mac
2023-05-01 15:41:16 : INFO  : cleartouchcollage : name: Collage, appName: Collage.app
2023-05-01 15:41:16.456 mdfind[69946:2560716] [UserQueryParser] Loading keywords and predicates for locale "en_US"
2023-05-01 15:41:16.456 mdfind[69946:2560716] [UserQueryParser] Loading keywords and predicates for locale "en"
2023-05-01 15:41:16.490 mdfind[69946:2560716] Couldn't determine the mapping between prefab keywords and predicates.
2023-05-01 15:41:16 : WARN  : cleartouchcollage : No previous app found
2023-05-01 15:41:16 : WARN  : cleartouchcollage : could not find Collage.app
2023-05-01 15:41:16 : INFO  : cleartouchcollage : appversion: 
2023-05-01 15:41:16 : INFO  : cleartouchcollage : Latest version of Collage is 5.5.0.3459
2023-05-01 15:41:16 : REQ   : cleartouchcollage : Downloading https://www.getcleartouch.com/download/collage-for-mac/?wpdmdl=412 to Collage.zip
2023-05-01 15:41:22 : REQ   : cleartouchcollage : no more blocking processes, continue with update
2023-05-01 15:41:22 : REQ   : cleartouchcollage : Installing Collage
2023-05-01 15:41:22 : INFO  : cleartouchcollage : Unzipping Collage.zip
2023-05-01 15:41:22 : INFO  : cleartouchcollage : found pkg: /var/folders/zz/zyxvpxvq6csfxvn_n0000000000000/T/tmp.YtTdWQAu/Collage_V5.5.0.3459 2/Collage.5.5.0.6803.pkg
2023-05-01 15:41:22 : INFO  : cleartouchcollage : Verifying: /var/folders/zz/zyxvpxvq6csfxvn_n0000000000000/T/tmp.YtTdWQAu/Collage_V5.5.0.3459 2/Collage.5.5.0.6803.pkg
2023-05-01 15:41:22 : INFO  : cleartouchcollage : Team ID: P76M9BE8DQ (expected: P76M9BE8DQ )
2023-05-01 15:41:22 : INFO  : cleartouchcollage : Installing /var/folders/zz/zyxvpxvq6csfxvn_n0000000000000/T/tmp.YtTdWQAu/Collage_V5.5.0.3459 2/Collage.5.5.0.6803.pkg to /
2023-05-01 15:41:26 : INFO  : cleartouchcollage : Finishing...
2023-05-01 15:41:29 : INFO  : cleartouchcollage : found packageID com.cvte.cleartouch.mac installed, version 5.5.0.6803
2023-05-01 15:41:29 : REQ   : cleartouchcollage : Installed Collage, version 5.5.0.6803
2023-05-01 15:41:29 : INFO  : cleartouchcollage : notifying
2023-05-01 15:41:30 : INFO  : cleartouchcollage : App not closed, so no reopen.
2023-05-01 15:41:30 : REQ   : cleartouchcollage : All done!
2023-05-01 15:41:30 : REQ   : cleartouchcollage : ################## End Installomator, exit code 0

scriptingosx commented:
getting a `Team IDs do not match!` error here?

duderin0 commented:
Amazing, they've changed the download already. One zip to hold 2 more zips, with either DMG or PKG inside the 2nd zip now. I only got "TeamIDs do not match!" when there was another pkg in the build folder from another assemble, since Installomator searches for one and all that gets unzipped is two more zips. Are there any methods for dealing with nested zips? I might reach out to the company to see if they could be a little more consistent with their download...

scriptingosx commented:
yeah, providing a dmg and pkg inside a zip is really showing off their ignorance of the way things work on macOS. "People have asked for pkg and a dmg and we have no idea how this work so we give them both! in a zip!" ... 🤦

duderin0 commented:
Well before I got around to contacting them, I found lower on the download page (https://www.getcleartouch.com/download/collage-for-mac/) they have the two zips separated and I've switched to xpath to target the one we need, so when/if they rename the file (again) hopefully it can still get picked up. Speaking of name, previously the only way to get the full version number was in the filename but now they've changed that to only the major version, so I've removed that option. I'll ask them if they can put the full version on the page and will update the label for it if they do, but its a small enough (10MB) download so I'm fixing and deploying it for now.

Output:
./assemble.sh cleartouchcollage
2023-05-30 09:03:22 : REQ   : cleartouchcollage : ################## Start Installomator v. 10.4beta, date 2023-05-30
2023-05-30 09:03:22 : INFO  : cleartouchcollage : ################## Version: 10.4beta
2023-05-30 09:03:22 : INFO  : cleartouchcollage : ################## Date: 2023-05-30
2023-05-30 09:03:22 : INFO  : cleartouchcollage : ################## cleartouchcollage
2023-05-30 09:03:22 : DEBUG : cleartouchcollage : DEBUG mode 1 enabled.
2023-05-30 09:03:23 : DEBUG : cleartouchcollage : name=Collage
2023-05-30 09:03:23 : DEBUG : cleartouchcollage : appName=
2023-05-30 09:03:23 : DEBUG : cleartouchcollage : type=pkgInZip
2023-05-30 09:03:23 : DEBUG : cleartouchcollage : archiveName=
2023-05-30 09:03:23 : DEBUG : cleartouchcollage : downloadURL=https://www.getcleartouch.com/download/collage-for-mac/?wpdmdl=412&refresh=6475e837add911685448759&ind=1683902820893&filename=1683902820wpdm_Collage%20v5.5%20PKG.zip
2023-05-30 09:03:23 : DEBUG : cleartouchcollage : curlOptions=
2023-05-30 09:03:23 : DEBUG : cleartouchcollage : appNewVersion=
2023-05-30 09:03:23 : DEBUG : cleartouchcollage : appCustomVersion function: Not defined
2023-05-30 09:03:23 : DEBUG : cleartouchcollage : versionKey=CFBundleShortVersionString
2023-05-30 09:03:23 : DEBUG : cleartouchcollage : packageID=com.cvte.cleartouch.mac
2023-05-30 09:03:23 : DEBUG : cleartouchcollage : pkgName=
2023-05-30 09:03:23 : DEBUG : cleartouchcollage : choiceChangesXML=
2023-05-30 09:03:23 : DEBUG : cleartouchcollage : expectedTeamID=P76M9BE8DQ
2023-05-30 09:03:23 : DEBUG : cleartouchcollage : blockingProcesses=
2023-05-30 09:03:23 : DEBUG : cleartouchcollage : installerTool=
2023-05-30 09:03:23 : DEBUG : cleartouchcollage : CLIInstaller=
2023-05-30 09:03:23 : DEBUG : cleartouchcollage : CLIArguments=
2023-05-30 09:03:23 : DEBUG : cleartouchcollage : updateTool=
2023-05-30 09:03:23 : DEBUG : cleartouchcollage : updateToolArguments=
2023-05-30 09:03:23 : DEBUG : cleartouchcollage : updateToolRunAsCurrentUser=
2023-05-30 09:03:23 : INFO  : cleartouchcollage : BLOCKING_PROCESS_ACTION=tell_user
2023-05-30 09:03:23 : INFO  : cleartouchcollage : NOTIFY=success
2023-05-30 09:03:23 : INFO  : cleartouchcollage : LOGGING=DEBUG
2023-05-30 09:03:23 : INFO  : cleartouchcollage : LOGO=/System/Applications/App Store.app/Contents/Resources/AppIcon.icns
2023-05-30 09:03:23 : INFO  : cleartouchcollage : Label type: pkgInZip
2023-05-30 09:03:23 : INFO  : cleartouchcollage : archiveName: Collage.zip
2023-05-30 09:03:23 : INFO  : cleartouchcollage : no blocking processes defined, using Collage as default
2023-05-30 09:03:23 : DEBUG : cleartouchcollage : Changing directory to /Users/*/Documents/GitHub/Installomator/build
2023-05-30 09:03:23 : INFO  : cleartouchcollage : found packageID com.cvte.cleartouch.mac installed, version 5.4.2.6609
2023-05-30 09:03:23 : INFO  : cleartouchcollage : appversion: 5.4.2.6609
2023-05-30 09:03:23 : INFO  : cleartouchcollage : Latest version not specified.
2023-05-30 09:03:23 : REQ   : cleartouchcollage : Downloading https://www.getcleartouch.com/download/collage-for-mac/?wpdmdl=412&refresh=6475e837add911685448759&ind=1683902820893&filename=1683902820wpdm_Collage%20v5.5%20PKG.zip to Collage.zip
2023-05-30 09:03:23 : DEBUG : cleartouchcollage : No Dialog connection, just download
2023-05-30 09:03:26 : DEBUG : cleartouchcollage : File list: -rw-r--r--  1 admin  2103187081   9.5M May 30 09:03 Collage.zip
2023-05-30 09:03:26 : DEBUG : cleartouchcollage : File type: Collage.zip: Zip archive data, at least v2.0 to extract, compression method=deflate
2023-05-30 09:03:26 : DEBUG : cleartouchcollage : curl output was:
*   Trying 35.244.153.44:443...
* Connected to [www.getcleartouch.com](http://www.getcleartouch.com/) (35.244.153.44) port 443 (#0)
* ALPN: offers h2,http/1.1
* (304) (OUT), TLS handshake, Client hello (1):
} [326 bytes data]
*  CAfile: /etc/ssl/cert.pem
*  CApath: none
* (304) (IN), TLS handshake, Server hello (2):
{ [88 bytes data]
* (304) (OUT), TLS handshake, Client hello (1):
} [391 bytes data]
* (304) (IN), TLS handshake, Server hello (2):
{ [187 bytes data]
* (304) (IN), TLS handshake, Unknown (8):
{ [19 bytes data]
* (304) (IN), TLS handshake, Certificate (11):
{ [4056 bytes data]
* (304) (IN), TLS handshake, CERT verify (15):
{ [264 bytes data]
* (304) (IN), TLS handshake, Finished (20):
{ [52 bytes data]
* (304) (OUT), TLS handshake, Finished (20):
} [52 bytes data]
* SSL connection using TLSv1.3 / AEAD-AES256-GCM-SHA384
* ALPN: server accepted h2
* Server certificate:
*  subject: CN=*.getcleartouch.com
*  start date: Apr 15 02:40:12 2023 GMT
*  expire date: Jul 14 02:40:11 2023 GMT
*  subjectAltName: host "[www.getcleartouch.com](http://www.getcleartouch.com/)" matched cert's "*.getcleartouch.com"
*  issuer: C=US; O=Let's Encrypt; CN=R3
*  SSL certificate verify ok.
* using HTTP/2
* h2h3 [:method: GET]
* h2h3 [:path: /download/collage-for-mac/?wpdmdl=412&refresh=6475e837add911685448759&ind=1683902820893&filename=1683902820wpdm_Collage%20v5.5%20PKG.zip]
* h2h3 [:scheme: https]
* h2h3 [:authority: [www.getcleartouch.com](http://www.getcleartouch.com/)]
* h2h3 [user-agent: curl/7.88.1]
* h2h3 [accept: */*]
* Using Stream ID: 1 (easy handle 0x14d813400)
> GET /download/collage-for-mac/?wpdmdl=412&refresh=6475e837add911685448759&ind=1683902820893&filename=1683902820wpdm_Collage%20v5.5%20PKG.zip HTTP/2
> Host: [www.getcleartouch.com](http://www.getcleartouch.com/)
> user-agent: curl/7.88.1
> accept: */*
>
< HTTP/2 200
< server: nginx
< date: Tue, 30 May 2023 16:03:24 GMT
< content-type: application/zip
< content-length: 10005866
< expires: Wed, 11 Jan 1984 05:00:00 GMT
< cache-control: no-cache, must-revalidate, max-age=0
< x-robots-tag: noindex, nofollow
< robots: none
< content-description: File Transfer
< content-disposition: attachment;filename="Collage%20v5.5%20PKG.zip"
< content-transfer-encoding: binary
< set-cookie: __wpdm_client=5c8ac4f1c7d5b638bf2d149f1e7c892a; path=/; secure; HttpOnly
< x-httpd: 1
< x-cdn-c: all
< x-sg-cdn: 1
< x-proxy-cache: MISS
< x-proxy-cache-info: 0 NC:000000 UP:SKIP_CACHE_SET_COOKIE
< host-header: 8441280b0c35cbc1147f8ba998a563a7
<
{ [15906 bytes data]
* Connection #0 to host [www.getcleartouch.com](http://www.getcleartouch.com/) left intact

2023-05-30 09:03:26 : DEBUG : cleartouchcollage : DEBUG mode 1, not checking for blocking processes
2023-05-30 09:03:26 : REQ   : cleartouchcollage : Installing Collage
2023-05-30 09:03:26 : INFO  : cleartouchcollage : Unzipping Collage.zip
2023-05-30 09:03:26 : DEBUG : cleartouchcollage : Found pkg(s):
/Users/admin/Documents/GitHub/Installomator/build/Collage v5.5 PKG/Collage.5.5.0.6830.pkg
2023-05-30 09:03:26 : INFO  : cleartouchcollage : found pkg: /Users/admin/Documents/GitHub/Installomator/build/Collage v5.5 PKG/Collage.5.5.0.6830.pkg
2023-05-30 09:03:26 : INFO  : cleartouchcollage : Verifying: /Users/admin/Documents/GitHub/Installomator/build/Collage v5.5 PKG/Collage.5.5.0.6830.pkg
2023-05-30 09:03:26 : DEBUG : cleartouchcollage : File list: -rw-r--r--  1 admin  2103187081   9.6M May  8 07:45 /Users/admin/Documents/GitHub/Installomator/build/Collage v5.5 PKG/Collage.5.5.0.6830.pkg
2023-05-30 09:03:26 : DEBUG : cleartouchcollage : File type: /Users/admin/Documents/GitHub/Installomator/build/Collage v5.5 PKG/Collage.5.5.0.6830.pkg: xar archive compressed TOC: 4952, SHA-1 checksum
2023-05-30 09:03:26 : DEBUG : cleartouchcollage : spctlOut is /Users/admin/Documents/GitHub/Installomator/build/Collage v5.5 PKG/Collage.5.5.0.6830.pkg: accepted
2023-05-30 09:03:26 : DEBUG : cleartouchcollage : source=Notarized Developer ID
2023-05-30 09:03:26 : DEBUG : cleartouchcollage : origin=Developer ID Installer: Guangzhou Shirui Electronics Co., Ltd. (P76M9BE8DQ)
2023-05-30 09:03:26 : INFO  : cleartouchcollage : Team ID: P76M9BE8DQ (expected: P76M9BE8DQ )
2023-05-30 09:03:26 : INFO  : cleartouchcollage : Checking package version.
2023-05-30 09:03:27 : INFO  : cleartouchcollage : Downloaded package com.cvte.cleartouch.mac version 5.5.0.6830
2023-05-30 09:03:27 : DEBUG : cleartouchcollage : DEBUG enabled, skipping installation
2023-05-30 09:03:27 : INFO  : cleartouchcollage : Finishing...
2023-05-30 09:03:30 : INFO  : cleartouchcollage : found packageID com.cvte.cleartouch.mac installed, version 5.4.2.6609
2023-05-30 09:03:30 : REQ   : cleartouchcollage : Installed Collage, version 5.4.2.6609
2023-05-30 09:03:30 : INFO  : cleartouchcollage : notifying
2023-05-30 09:03:30 : DEBUG : cleartouchcollage : DEBUG mode 1, not reopening anything
2023-05-30 09:03:30 : REQ   : cleartouchcollage : All done!
2023-05-30 09:03:30 : REQ   : cleartouchcollage : ################## End Installomator, exit code 0

scriptingosx commented:
Awesome! great sleuthing!

scriptingosx commented:
Thank you!
